spec/datadog/integration_spec.rb

```diff
@@ -23,22 +23,30 @@ def wait_for_tracer_sent
 
     context 'for threads' do
       let(:original_threads) { Thread.list }
-      let!(:original_threads_inspect) { inspect_threads(original_threads) } # Store result as stack trace will change
-      let(:threads) { Thread.list }
+      let(:start_tracer) do
+        Datadog::Tracing.trace('test.op') {}
+
+        new_threads = Thread.list - original_threads
```

Comment:
Be careful with the lazy initialization of `let`s. Since `original_threads` is only referenced after starting the tracer, this line is the equivalent of `new_threads = Thread.list - Thread.list`.